lib/vagrant/util/downloader.rb

```diff
@@ -203,9 +203,13 @@ def execute_curl(options, subprocess_options, &data_proc)
           @logger.warn("Downloader exit code: #{result.exit_code}")
           parts    = result.stderr.split(/\n*curl:\s+\(\d+\)\s*/, 2)
           parts[1] ||= ""
-          raise Errors::DownloaderError,
-            code: result.exit_code,
-            message: parts[1].chomp
+          if parts[1].include? "416"
+            # All good actually. 416 means there is no mory bytes to download
```

Comment:
@glennpratt This is true. You can refer to HTTP response codes specification.

416 response code is returned when "requested starting byte is out of range". In this case client sends an offset (in bytes) starting with which it wants to receive data, if http server supports this mode, then it 1 - checks the size of files and if the size is more than specified offset then 2 - seek to the offset and starts streaming from it. This is how "resume download" implemented in all cases. 416 code is returned when 1 is not satisfiable (meaning specified offset is out of byte range of actual file size). If you get 416 on "not completely" downloaded file, this only means that client sent offset that is outside of range - either client bug or client received garbage that made file size same/bigger than the one on the server.

I am just following specifications of HTTP(S) protocol here.



> s/mory/more

A minor typo ;)